hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -106,7 +106,8 @@ long getLastCallId() {
     public boolean register(Invocation invocation) {
         final long callId;
         try {
-            callId = callIdSequence.next(invocation.op.isUrgent());
+            boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
```

Comment:
here is a scenario to show at most + 1 retry can be executed. 

```
occupied slots: N

retry queue: R1, R2, R3

1) R1 deregister: N -> N - 1
2) new non-retry invocation: N - 1 -> N
3) R1 register: N -> N + 1
4) R2 deregister: N + 1 -> N. A new non-retry invocation cannot start since all slots are occupied. 
5) R2 register: N -> N + 1
```